### PR TITLE
Teardown lambda

### PR DESCRIPTION
**Pull request by PoppleJIW**

### CODE REVIEW

File: src/common/common.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    if "RDS_PASSWORD_SECRET_NAME" in os.environ:
+        _args["rds_password_secret_name"] = os.environ["RDS_PASSWORD_SECRET_NAME"]
+
+    if "RECONCILER_MAXIMUM_AGE_SCALE" in os.environ:
+        _args["reconciler_maximum_age_scale"] = os.environ[
+            "RECONCILER_MAXIMUM_AGE_SCALE"
+        ]
+
+    if "RECONCILER_MAXIMUM_AGE_UNIT" in os.environ:
+        _args["reconciler_maximum_age_unit"] = os.environ["RECONCILER_MAXIMUM_AGE_UNIT"]
```
Comment (steveburton4) [inline src/common/common.py:93]: These aren't needed in this lambda I think

Comment (steveburton4) [reply src/common/common.py:93]: Can you check the ones above are all used as well?

Comment (steveburton4) [reply src/common/common.py:93]: If you remove any, remember to check and remove from the required list below too.

File: src/common/common.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        )
+
+    # Validate table name
+    if (
+        "table-name" in _args
+        and _args["table-name"].upper() not in database.Table.__members__
+    ):
+        raise ValueError(
+            f"ValueError: table-name {_args['table-name'].upper()} is invalid or not supported"
+        )
```
Comment (steveburton4) [inline src/common/common.py:123]: This is only needed if we pass in table name? Do we do this? If so, cool, leave it in.

File: src/common/database.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import mysql.connector
+import mysql.connector.pooling
+
+logger = logging.getLogger(__name__)
+
+
+class Table(Enum):
+    UCFS = "ucfs"
+    EQUALITIES = "equalities"
+    AUDIT = "audit"
```
Comment (steveburton4) [inline src/common/database.py:15]: I don't think we need this as we aren't targetting specific tables?

File: src/teardown_lambda/teardown.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+from common import common, database
+
+logger = None
+
+# Copy tables
+copy_queries = [
+    # Claimant
+    """
+    CREATE TABLE IF NOT EXISTS claimant_stage LIKE claimant_old;
+    INSERT INTO claimant_stage (id, data) SELECT id, data from claimant_old;
```
Comment (steveburton4) [inline src/teardown_lambda/teardown.py:13]: I would probably create a variable named `sql_statement` with `CREATE TABLE IF NOT EXISTS [db_name]_stage LIKE [db_name]_old;
    INSERT INTO [db_name]_stage (id, data) SELECT id, data from [db_name]_old;` in and then when creating this array do something like `sql_statement.replace("[db_name]", "claimant")` and so on for the three, to allow one place to change the statement in the future.

File: src/teardown_lambda/teardown.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+        for query in copy_queries:
+            database.execute_multiple_statements(query, connection)
+
+        database.execute_multiple_statements(drop_query, connection)
+        database.execute_statement(rename_query, connection)
+
+        return 200
+    except Exception as e:
+        logger.error(e)
```
Comment (steveburton4) [inline src/teardown_lambda/teardown.py:60]: We can log the exception but we should log a readable error string first, like `logger.error("Could not execute one or more sql statements")`.

File: src/common/common.py (hunk 1 above)
Comment (PoppleJIW) [reply src/common/common.py]: Removed this snippet, couldn't find anything in the required section below for those 2 keys.

File: src/common/common.py (hunk 2 above)
Comment (PoppleJIW) [reply src/common/common.py]: Removed this snippet.

File: src/common/database.py (hunk 3 above)
Comment (PoppleJIW) [reply src/common/database.py]: Removed this snippet as the only usage of the class was removed.

**Review by steveburton4 — DISMISSED** [review]

**Review by steveburton4 — APPROVED** [review]

**Pull request merged**